chain/chain/src/store.rs

```diff
@@ -2368,9 +2422,31 @@ impl<'a> ChainStoreUpdate<'a> {
                 receipt,
             )?;
         }
+        let mut block_hash_to_outcomes: HashMap<CryptoHash, HashSet<CryptoHash>> = HashMap::new();
         for (hash, outcome) in self.chain_store_cache_update.outcomes.iter() {
+            match block_hash_to_outcomes.entry(outcome.block_hash) {
+                Entry::Occupied(mut entry) => {
+                    entry.get_mut().insert(outcome.id().clone());
+                }
+                Entry::Vacant(entry) => {
+                    let mut hash_set = match self
+                        .chain_store
+                        .get_all_block_hash_to_outcomes(&outcome.block_hash)
+                    {
+                        Ok(hash_set) => hash_set.clone(),
+                        Err(_) => HashSet::new(),
+                    };
+                    hash_set.insert(outcome.id().clone());
+                    entry.insert(hash_set);
+                }
+            };
             store_update.set_ser(ColTransactionResult, hash.as_ref(), outcome)?;
         }
+        for (block_hash, hash_set) in block_hash_to_outcomes {
+            store_update
+                .set_ser(ColOutcomesByBlockHash, block_hash.as_ref(), &hash_set)
+                .map_err::<Error, _>(|e| e.into())?;
```

Comment:
`?` should call `.into()` automatically, so this `.map_err` seems to be redundant, doesn't it?

Reply:
++